### PR TITLE
Added deep vs shallow DirToJSON

### PR DESCRIPTION
**Pull request by cftorres**
@abbytiz @aaron-prindle @nkubala

### CODE REVIEW

File: iDiff/utils/tar_utils.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func ImageToDir(img string) (string, string, error) {
 	os.Remove(tarPath)
 	path := strings.TrimSuffix(tarPath, filepath.Ext(tarPath))
 	jsonPath := path + ".json"
-	err = DirToJSON(path, jsonPath)
+	err = DirToJSON(path, jsonPath, false)
```
Comment (abbytiz) [inline iDiff/utils/tar_utils.go]: Do we always want this to be false?  Would it be useful to add the boolean as an argument for ImageToDir so for a specific differ we can control the shallow or deep output?

**Comment by nkubala** [comment]
what are the cases where we would want shallow vs deep directory walking?

**Comment by cftorres** [comment]
we were thinking for optimization purposes if we ever don't need to crawl all the way down a directory structure, such as if we simply wanted to examine which layers had changed in the image tar root, we only would need to look at the first level in the directory hierarchy

**Comment by nkubala** [comment]
alright that makes sense. let's just make sure that whenever we don't go all the way down a directory, it's for a good reason. don't want to be ignoring output that users are expecting to see. it would probably be useful to add some e2e tests that make sure this is being used correctly in different places (i.e. a normal diff should be doing a deep directory walk, whereas a "root" diff should only be doing a shallow walk).

**Pull request merged**